buildenv

```diff
@@ -1,84 +1,192 @@
   cd libs/multiprecision
-  git apply $ZOPEN_ROOT/patches/multiprecision.diff
+  git apply $ZOPEN_ROOT/boostPatches/multiprecision.patch
   cd -
 
   # thread directory  patch -> located at ./boost/libs/thread
-  echo "In ZOPEN_INIT, applying patch to $ZOPEN_ROOT/patches/thread.diff"
+  echo "In ZOPEN_INIT, applying patch to $ZOPEN_ROOT/boostPatches/thread.patch"
   cd libs/thread
-  git apply $ZOPEN_ROOT/patches/thread.diff
+  git apply $ZOPEN_ROOT/boostPatches/thread.hpp.patch
+  git apply $ZOPEN_ROOT/boostPatches/thread_data.hpp.patch
+  git apply $ZOPEN_ROOT/boostPatches/pthread_helpers.hpp.patch
   cd -
 
   # build directory  patch -> located at ./boost/tools/build
-  echo "In ZOPEN_INIT, applying patch to $ZOPEN_ROOT/patches/build.diff"
-  cd tools/build
-  git apply $ZOPEN_ROOT/patches/build.diff
-  #git apply $ZOPEN_ROOT/patches/build_without_clang-linux.diff
+  echo "In ZOPEN_INIT, applying patch to $ZOPEN_ROOT/boostPatches/build.patch"
+  cd tools/build/
+  git apply $ZOPEN_ROOT/boostPatches/src_engine.patch
+  git apply $ZOPEN_ROOT/boostPatches/gcc.jam.patch
+  git apply $ZOPEN_ROOT/boostPatches/toolset.jam.patch
+  git apply $ZOPEN_ROOT/boostPatches/clang.jam.patch
+  git apply $ZOPEN_ROOT/boostPatches/intel-linux.jam.patch
+  git apply $ZOPEN_ROOT/boostPatches/pathscale.jam.patch
+  git apply $ZOPEN_ROOT/boostPatches/pgi.jam.patch
+  git apply $ZOPEN_ROOT/boostPatches/vacpp.jam.patch
+  git apply $ZOPEN_ROOT/boostPatches/xlcpp.jam.patch
+  git apply $ZOPEN_ROOT/boostPatches/clang-linux-3.9.0.py.patch
+  git apply $ZOPEN_ROOT/boostPatches/gcc-4.8.3-linux.py.patch
+  cd -
+  
+  cd ./libs/iostreams
+  git apply $ZOPEN_ROOT/boostPatches/iostreams.Jamfile.v2.patch
+  cd -
+
+  cd ./libs/chrono
+  git apply $ZOPEN_ROOT/boostPatches/chrono.hpp.patch
+  git apply $ZOPEN_ROOT/boostPatches/chrono_io.hpp.patch
+  git apply $ZOPEN_ROOT/boostPatches/time_point_io.hpp.patch
+  git apply $ZOPEN_ROOT/boostPatches/runtime_resolution.cpp.patch
+  git apply $ZOPEN_ROOT/boostPatches/time_point_io.hpp.patch
+  git apply $ZOPEN_ROOT/boostPatches/duration_duration_fail.cpp.patch
+  git apply $ZOPEN_ROOT/boostPatches/system_clocks.hpp.patch
+  cd -
+
+  cd ./libs/test
+  git apply $ZOPEN_ROOT/boostPatches/execution_monitor.ipp.patch
+  cd -
+
+  cd ./libs/filesystem
+  git apply $ZOPEN_ROOT/boostPatches/directory.cpp.patch 
   cd -
+
+  echo `pwd`
 }
 
 zopen_pre_configure()
 {
-  sed -i 's/using clang ;/using clang : : "ibm-clang++64" : <cxxflags>" -DNDEBUG -march=arch10 -O3 -m64 -D_XOPEN_SOURCE=600 -D_UNIX03_THREADS -D_POSIX_THREADS -D_OPEN_SYS_FILE_EXT -fasm -fzos-le-char-mode=ascii -fno-short-enums" ;/g' project-config.jam 
+  var="using clang : clang++ : <compileflags>-DNDEBUG -march=arch10 -O3 -m64 -D_XOPEN_SOURCE=600 -D_UNIX03_THREADS -D_POSIX_THREADS -D_OPEN_SYS_FILE_EXT -fasm -fzos-le-char-mode=ascii -fno-short-enums '\\${ZOPEN_EXTRA_CPPFLAGS}' '\\${ZOPEN_EXTRA_CFLAGS}' '\\${ZOPEN_EXTRA_CXXFLAGS}' : 14.0.0 ;"
```

Comment:
Can we remove these hardcoded flags?

Reply:
Done, retained few as they werent part of the default flags. Thank you!